Nice looks good to me, full build fails because of expected and know failure:
```2026-08-28T10:02:01.4910735Z [32m2026-08-28 10:02:01,490 docker_builder.py:108        INFO  Docker: [INFO] ------------------------------------------------------------------------[0m
2026-08-28T10:02:01.4912787Z [32m2026-08-28 10:02:01,490 docker_builder.py:108        INFO  Docker: [INFO] BUILD FAILURE[0m
2026-08-28T10:02:01.4915171Z [32m2026-08-28 10:02:01,491 docker_builder.py:108        INFO  Docker: [INFO] ------------------------------------------------------------------------[0m
2026-08-28T10:02:01.4919336Z [32m2026-08-28 10:02:01,491 docker_builder.py:108        INFO  Docker: [INFO] Total time:  01:43 min[0m
2026-08-28T10:02:01.4922305Z [32m2026-08-28 10:02:01,491 docker_builder.py:108        INFO  Docker: [INFO] Finished at: 2026-08-28T10:02:01Z[0m
2026-08-28T10:02:01.4924955Z [32m2026-08-28 10:02:01,492 docker_builder.py:108        INFO  Docker: [INFO] ------------------------------------------------------------------------[0m
2026-08-28T10:02:01.4937593Z [32m2026-08-28 10:02:01,493 docker_builder.py:108        INFO  Docker: [ERROR] Failed to execute goal com.redhat.quarkus.platform:quarkus-maven-plugin:3.33.2.SP2-redhat-00002:go-offline (default-cli) on project serverless-workflow-project: Execution default-cli of goal com.redhat.quarkus.platform:quarkus-maven-plugin:3.33.2.SP2-redhat-00002:go-offline failed: The following errors were encountered while processing Quarkus application dependencies:[0m
2026-08-28T10:02:01.4942966Z [32m2026-08-28 10:02:01,493 docker_builder.py:108        INFO  Docker: [ERROR] 1) Failed to resolve artifact io.quarkus:quarkus-jdbc-postgresql:jar:3.33.2.SP2-redhat-00002[0m
2026-08-28T10:02:01.4946257Z [32m2026-08-28 10:02:01,493 docker_builder.py:108        INFO  Docker: [ERROR]   at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver.resolve(ApplicationDependencyResolver.java:959)[0m
2026-08-28T10:02:01.4955568Z [32m2026-08-28 10:02:01,493 docker_builder.py:108        INFO  Docker: [ERROR]   ...[0m
2026-08-28T10:02:01.4958366Z [32m2026-08-28 10:02:01,493 docker_builder.py:108        INFO  Docker: [ERROR] 2) Failed to resolve artifact io.quarkus:quarkus-jdbc-postgresql-deployment:jar:3.33.2.SP2-redhat-00002[0m
2026-08-28T10:02:01.4970403Z [32m2026-08-28 10:02:01,493 docker_builder.py:108        INFO  Docker: [0m
2026-08-28T10:02:01.4972209Z [32m2026-08-28 10:02:01,494 docker_builder.py:108        INFO  Docker: [ERROR]   at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver.resolve(ApplicationDependencyResolver.java:959)[0m
2026-08-28T10:02:01.4973415Z [32m2026-08-28 10:02:01,494 docker_builder.py:108        INFO  Docker: [ERROR]   ...[0m
2026-08-28T10:02:01.4974201Z [32m2026-08-28 10:02:01,494 docker_builder.py:108        INFO  Docker: [ERROR] -> [Help 1][0m
2026-08-28T10:02:01.4975249Z [32m2026-08-28 10:02:01,494 docker_builder.py:108        INFO  Docker: [ERROR][0m
2026-08-28T10:02:01.4976300Z [32m2026-08-28 10:02:01,494 docker_builder.py:108        INFO  Docker: [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[0m
2026-08-28T10:02:01.4977495Z [32m2026-08-28 10:02:01,495 docker_builder.py:108        INFO  Docker: [ERROR] Re-run Maven using the -X switch to enable full debug logging.[0m
2026-08-28T10:02:01.4978358Z [32m2026-08-28 10:02:01,495 docker_builder.py:108        INFO  Docker: [ERROR][0m
2026-08-28T10:02:01.4979543Z [32m2026-08-28 10:02:01,495 docker_builder.py:108        INFO  Docker: [ERROR] For more information about the errors and possible solutions, please read the following articles:[0m
2026-08-28T10:02:01.4980959Z [32m2026-08-28 10:02:01,496 docker_builder.py:108        INFO  Docker: [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException[0m
2026-08-28T10:02:01.6930880Z [31m2026-08-28 10:02:01,692 docker_builder.py:140        ERROR You can look inside the failed image by running 'docker run --rm -ti e6b7a2f6a054 bash'[0m
2026-08-28T10:02:01.6951654Z [31m2026-08-28 10:02:01,694 docker_builder.py:153        ERROR Image build failed: 'The command 'sh -x /tmp/scripts/org.kie.sonataflow.common.build/configure.sh' returned a non-zero code: 1'[0m
2026-08-28T10:02:01.6953642Z [31m2026-08-28 10:02:01,694 cli.py:565        ERROR Image build failed, see logs above.[0m```

--- a/.github/actions/bootstrap/action.yml
+++ b/.github/actions/bootstrap/action.yml
@@ -45,6 +45,38 @@ runs:
         cd ${{ inputs.working_dir }}
         df -h .
 
+    # Disabled: local maven cache for coordinated cross-repository PRs.
+    #
+    # Purpose: when a single change spans kie-tools and its upstream deps
+    # (drools / kogito-runtimes / kogito-apps), this step builds those deps
+    # from the PR author's own forks, using the PR head branch name, so the
+    # kie-tools PR is validated against the matching in-progress upstream
+    # change instead of the default kiegroup branch.
+    #
+    # Why it is disabled here:
+    #   - This composite action is currently used only by osl_sync_main.yml,
+    #     whose triggers are `schedule` and `workflow_dispatch` only (never
+    #     `pull_request`), so `if: github.event_name == 'pull_request'` can
+    #     never be true and the step is unreachable.
+    #   - The owner forks it points at (kubesmarts/{drools,kogito-runtimes,
+    #     kogito-apps}) do not exist, so setup-maven-cache would only fall
+    #     back to kiegroup and produce the same result as the non-PR step
+    #     below.
+    #
+    # Kept as a template: re-enable this block (and point the *_REPO_URL
+    # inputs at real forks) if a `pull_request`-triggered workflow ever
+    # consumes this action.
+    #
+    # - name: Setup local maven cache for PR
+    #   id: setup_maven_cache_pr
+    #   if: "github.event_name == 'pull_request'"
+    #   uses: ./.github/actions/setup-maven-cache
+    #   with:
+    #     DEPS_BRANCHNAME: ${{ github.head_ref }}
+    #     DROOLS_REPO_URL: "https://github.com/${{ github.event.pull_request.head.repo.owner.login }}/drools"
+    #     KOGITO_RUNTIMES_REPO_URL: "https://github.com/${{ github.event.pull_request.head.repo.owner.login }}/kogito-runtimes"
+    #     KOGITO_APPS_REPO_URL: "https://github.com/${{ github.event.pull_request.head.repo.owner.login }}/kogito-apps"
+
     - name: Setup local maven cache for non-pull request runs
       id: setup_maven_cache_non_pr
       if: "github.event_name != 'pull_request'"

